buildSrc/src/main/kotlin/androidx/build/ftl/GCloudCLIWrapper.kt

```diff
@@ -110,6 +162,95 @@ internal class GCloudCLIWrapper(
             private val SUCCESS_OUTCOMES = listOf("passed", "flaky")
         }
     }
+
+    /**
+     * Parameters for invoking a test on the Firebase Test Lab
+     */
+    internal data class RunTestParameters(
+        /**
+         * The project path for which we are executing the tests for.
+         */
+        val projectPath: String,
+        /**
+         * The tested APK file
+         */
+        val testedApk: File,
+        /**
+         * The test APK file which includes the instrumentation tests
+         */
+        val testApk: File,
+        /**
+         * The name of the GS bucket to save the results
+         */
+        val bucketName: String = DEFAULT_BUCKET_NAME,
+        /**
+         * The GS Bucket directory where the results will be saved
+         */
+        val resultsBucketDir: String = makeResultsDir(projectPath),
+        /**
+         * The local folder where we will download the test results
+         */
+        val resultsLocalDir: File,
+    ) {
+
+        /**
+         * Returns the path to the Google Cloud bucket where the test run results are saved
+         */
+        fun cloudBucketPath(): String {
+            return "gs://$bucketName/$resultsBucketDir"
+        }
+
+        companion object {
+            const val DEFAULT_BUCKET_NAME = "androidx-ftl-test-results"
+
+            /**
+             * The file into which the result of the gcloud command will be written.
+             */
+            const val TEST_OUTPUT_FILE_NAME = "testResults.json"
+
+            /**
+             * Generates a folder for test results.
+             *
+             * If run on Github Actions CI, this method will use the environment variables to
+             * create a unique path for the action.
+             * If run locally, this will create a random UUID for the folder.
+             */
+            fun makeResultsDir(
```

Comment:
Mark private. Additionally, should these just be private top level functions in this file?

Reply:
made it private. I'm always unsure where to scope them but it is in the companion object already and it is scoped to the parameters class so maybe that is OK?